### PR TITLE
Add aggregation parsing to Elasticsearch Filebeat producer

### PR DESCRIPTION
**Pull request by markchalloner**

### CODE REVIEW

File: producers/elasticsearch_filebeat/types/elasticsearch-filebeat-issue.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package types
 
 type ElasticSearchFilebeatResult struct {
 	Hits ElasticSearchFilebeatHits `json:"hits"`
+	Aggregations ElasticSearchFilebeatAggregations `json:"aggregations"`
```
Comment (chrisnovakovic) [inline producers/elasticsearch_filebeat/types/elasticsearch-filebeat-issue.go:5]: nit: no column alignment

```
Hits         ElasticSearchFilebeatHits         `json:"hits"`
Aggregations ElasticSearchFilebeatAggregations `json:"aggregations"`
```

File: producers/elasticsearch_filebeat/types/elasticsearch-filebeat-issue.go [hunk 2]
```diff
@@ -21,5 +22,23 @@ type ElasticSearchFilebeatSource struct {
...
+	Aggregation ElasticSearchFilebeatAggregation `json:"aggregation"`
+}
+
+type ElasticSearchFilebeatAggregation struct {
+	Buckets []ElasticSearchFilebeatBucket `json:"buckets"`
+}
+
+// Note that the metric field name depends on the request.
+type ElasticSearchFilebeatBucket struct {
+	Name    string 					   `json:"key"`
```
Comment (chrisnovakovic) [inline producers/elasticsearch_filebeat/types/elasticsearch-filebeat-issue.go:38]: nit: incorrect column alignment

**Review by chrisnovakovic — CHANGES_REQUESTED** [review]
Looks good 🙂 Just a couple of nits...

**Comment by markchalloner** [comment]
Fixed some whitespace issues in the test data. Diff without whitespace:

https://github.com/thought-machine/dracon/pull/86/files?w=1

File: producers/elasticsearch_filebeat/types/elasticsearch-filebeat-issue.go [hunk 3]
```diff
@@ -1,25 +1,44 @@
...
+    Aggregation ElasticSearchFilebeatAggregation `json:"aggregation"`
+}
+
+type ElasticSearchFilebeatAggregation struct {
+    Buckets []ElasticSearchFilebeatBucket `json:"buckets"`
+}
+
+// Note that the metric field name depends on the request.
+type ElasticSearchFilebeatBucket struct {
+    Name    string                     `json:"key"`
```
Comment (chrisnovakovic) [inline producers/elasticsearch_filebeat/types/elasticsearch-filebeat-issue.go:38]: nit: still off by one here (between the field name and the type)

**Review by chrisnovakovic — CHANGES_REQUESTED** [review]

**Review by chrisnovakovic — CHANGES_REQUESTED** [review]
One remaining nit...

**Comment by markchalloner** [comment]
ah sorry good eye, thought I had resolved all these, fixing

**Review by chrisnovakovic — APPROVED** [review]

**Pull request merged**